### PR TITLE
`.count` metric naming convention only applies to UpDownCounters

### PR DESCRIPTION
**Pull request by trask**
Fixes https://github.com/open-telemetry/opentelemetry-specification/issues/3457

(port of https://github.com/open-telemetry/opentelemetry-specification/pull/3493)

### CODE REVIEW

**Comment by trask** [comment]
pinging @reyang @jack-berg @jsuereth @jmacd @MrAlias @gouthamve who approved this same PR before the repo split

(and @joaopgrassi who commented on it)

**Review by gouthamve — APPROVED** [review]
LGTM

**Review by joaopgrassi — APPROVED** [review]
Approving to get this improvement in. I guess the final outcome of it will be when we have an agreement on https://github.com/open-telemetry/semantic-conventions/issues/50. 

Should we also port the old PRs [#3476](https://github.com/open-telemetry/opentelemetry-specification/pull/3476)/[#3477](https://github.com/open-telemetry/opentelemetry-specification/pull/3477) here? SO we can voice our preferences again over which approach?

**Review by reyang — APPROVED** [review]

File: specification/metrics/semantic_conventions/README.md [hunk 1]
```diff
@@ -125,6 +126,22 @@ to the processes then to represent the count of the processes we can have a metr
...
+This rule SHOULD only be applied to UpDownCounters, since (monotonic) Counters have
+`_total` appended to their names when they are
+[mapped to Prometheus](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/compatibility/prometheus_and_openmetrics.md#otlp-metric-points-to-prometheus),
+which would lead to `_count_total`.
+
+#### Do not use `total`
+
+Counters SHOULD NOT append `_total`  to their names. Counters have `_total` appended to their names when they are
+[mapped to Prometheus](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/compatibility/prometheus_and_openmetrics.md#otlp-metric-points-to-prometheus),
+which would lead to `_total_total`. The reason that the Prometheus mapping cannot
```
Comment (dashpole) [inline specification/metrics/semantic_conventions/README.md:138]: FYI many prometheus exporters will omit _total if the metric already ends in _total, so I don't think this is a compelling reason.  IMO the better reason not to add `_total` is that it implies that the metric is cumulative, when it isn't neccessarily.

File: specification/metrics/semantic_conventions/README.md [hunk 2]
```diff
@@ -125,6 +126,22 @@ to the processes then to represent the count of the processes we can have a metr
...
+suppress adding the duplicate `_total` is because then the mapping wouldn't be
```
Comment (dashpole) [inline specification/metrics/semantic_conventions/README.md:139]: The prometheus receiver likely won't remove _total suffixes going forward, so this won't apply.  See https://github.com/open-telemetry/wg-prometheus/issues/72.

Comment (dashpole) [reply specification/metrics/semantic_conventions/README.md:139]: If system.processes.count is an UpDownCounter (which I expect it should be), then it is represented as a gauge in Prometheus.  In that case, it won't have _total appended.

**Review by jsuereth — APPROVED** [review]

**Comment by reyang** [comment]
@trask would you resolve the merge conflicts? I think it's time to get this merged.

File: specification/metrics/semantic_conventions/README.md (hunk 2 above)
Comment (reyang) [reply specification/metrics/semantic_conventions/README.md]: @dashpole PTAL.

File: docs/general/metrics.md [hunk 3]
```diff
@@ -125,6 +126,22 @@ to the processes then to represent the count of the processes we can have a metr
...
+This rule SHOULD only be applied to UpDownCounters, since (monotonic) Counters have
+`_total` appended to their names when they are
+[mapped to Prometheus](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/compatibility/prometheus_and_openmetrics.md#otlp-metric-points-to-prometheus),
+which would lead to `_count_total`.
+
+#### Do not use `total`
+
+Counters SHOULD NOT append `_total`  to their names. Counters have `_total` appended to their names when they are
+[mapped to Prometheus](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/compatibility/prometheus_and_openmetrics.md#otlp-metric-points-to-prometheus),
+which would lead to `_total_total`. The reason that the Prometheus mapping cannot
```
Comment (dashpole) [inline docs/general/metrics.md:138]: The [prometheus mapping](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#sums) does suppress duplicate `_total`: `If the metric name for monotonic Sum metric points does not end in a suffix of _total a suffix of _total MUST be added by default`.

It isn't fully bidirectional.  When converting prom -> OTLP, we leave the `_total` suffix intact by default per [The metadata spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#metric-metadata): `The OpenMetrics MetricFamily Name MUST be added as the Name of the OTLP metric. By default, the name MUST be unaltered, but translation SHOULD provide configuration which, when enabled, removes type (e.g. _total) and unit (e.g. _seconds) suffixes.`

Comment (trask) [reply docs/general/metrics.md]: thanks, I've removed this part

**Review by MrAlias — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Pull request merged**